### ISSUE TITLE
SArray initializer should accept any iterable

### ISSUE THREAD

**Issue by nickjong** [apple/turicreate#915]
It currently fails on e.g. python3 dict_keys:
```
>>> dict = {'alpha': 0, 'beta':1}
>>> sa = tc.SArray(dict.keys())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jong/Source/turicreate/debug/src/unity/python/turicreate/data_structures/sarray.py", line 422, in __init__
    raise TypeError("Unexpected data source. " \
TypeError: Unexpected data source. Possible data source types are: list, numpy.ndarray, pandas.Series, and string(url)
```

**Comment by dhivyaaxim** [comment]
@znation  Please assign this ticket to me as I would like to try fixing it and contribute .Thanks in advance!

**Comment by TobyRoseman** [comment]
The fix for this is probably very similar to #2470.

**Comment by TobyRoseman** [comment]
This is fixed in TuriCreate 6.4
